src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { ErrorMessage, TodosContext } from '../TodosContext';
+import { updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const {
+    handleRemoveTodo,
+    setAlarm,
+    setIsTodoChange,
+    changingItems,
+    setChangingItems,
+    setTodos,
+  } = useContext(TodosContext);
+
+  const [isCompleted, setIsCompleted] = useState(completed);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [editedInput, setEditedInput] = useState(title);
+
+  const editedFormInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isUpdating && editedFormInput.current) {
+      editedFormInput.current.focus();
+    }
+  }, [isUpdating]);
+
+  useEffect(() => {
+    setIsCompleted(completed);
+  }, [completed]);
+
+  const handleUpdateTodo = (val: boolean | string) => {
+    let newTodo = todo;
+
+    if (typeof val === 'boolean') {
+      newTodo = {
+        ...todo,
+        completed: val,
+      };
+    }
+
+    if (typeof val === 'string' && val.trim() !== '') {
+      newTodo = {
+        ...todo,
+        title: val,
+      };
+    }
+
+    if (newTodo) {
+      setIsTodoChange(true);
+      setChangingItems(current => [...current, id]);
+      updateTodo(newTodo)
+        .then(() => {
+          setIsUpdating(false);
+          setIsTodoChange(false);
+          setTodos(currentTodos => currentTodos
+            .map(currTodo => {
+              return currTodo.id !== newTodo.id
+                ? currTodo
+                : newTodo;
+            }));
+
+          setIsCompleted(newTodo.completed);
+        })
+        .catch(() => {
+          setAlarm(ErrorMessage.isUnableUpdateTodo);
+          editedFormInput.current?.focus();
+        })
+        .finally(() => {
+          setChangingItems([]);
+        });
+    }
+  };
+
+  const handleUpdateTodoTitle = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedInputValue = editedInput.trim();
+
+    if (!trimmedInputValue) {
+      handleRemoveTodo(todo);
+    }
```

Comment:
```suggestion
    if (!trimmedInputValue) {
      handleRemoveTodo(todo);
      
      return;
    }
```
You have to return after this as in this case, it tries to update todo too and I get an error.
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/63243136/81371e94-a404-407f-b2a3-efea76a30bcf)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/63243136/5974f174-2556-48ed-a56d-ab25fb500929)